var/spack/repos/builtin/packages/hdf5/package.py

```diff
@@ -223,106 +230,94 @@ def libs(self):
-        extra_args += self.enable_or_disable('fortran')
-        extra_args += self.enable_or_disable('java')
-
-        api = self.spec.variants['api'].value
-        if api != 'none':
-            extra_args.append('--with-default-api-version=' + api)
-
-        if '+szip' in self.spec:
-            szip_spec = self.spec['szip']
-            # The configure script of HDF5 accepts a comma-separated tuple of
-            # two paths: the first one points to the directory with include
-            # files, the second one points to the directory with library files.
-            # If the second path is not specified, the configure script assumes
-            # that it equals to prefix/lib. However, the correct directory
-            # might be prefix/lib64. It is not a problem when the building is
-            # done with Spack's compiler wrapper but it makes the Libtool
-            # files (*.la) invalid, which makes it problematic to use the
-            # installed library outside of Spack environment.
-            extra_args.append('--with-szlib=%s,%s' %
-                              (szip_spec.headers.directories[0],
-                               szip_spec.libs.directories[0]))
+        args.append('-DALLOW_UNSUPPORTED=ON')
+        
+        spec = self.spec
+        if '+zlib' in spec:
+            args.append('-DHDF5_ENABLE_Z_LIB_SUPPORT:BOOL=ON')
+            args.append('-DZLIB_INCLUDE_DIR:PATH={0}'.format(spec['zlib'].prefix.include))
+            args.append('-DZLIB_DIR:PATH={0}'.format(spec['zlib'].prefix.lib))
         else:
-            extra_args.append('--without-szlib')
+            args.append('-DHDF5_ENABLE_Z_LIB_SUPPORT:BOOL=OFF')
 
-        if self.spec.satisfies('@1.10:'):
-            if '+debug' in self.spec:
-                extra_args.append('--enable-build-mode=debug')
-            else:
-                extra_args.append('--enable-build-mode=production')
+        if '+mpi' in spec:
+            args.append('-DHDF5_ENABLE_PARALLEL=ON')
+            args.append('-DCMAKE_C_COMPILER={0}'.format(spec['mpi'].mpicc))
+            args.append('-DCMAKE_CXX_COMPILER={0}'.format(spec['mpi'].mpicxx))
+            args.append('-DCMAKE_Fortran_COMPILER={0}'.format(spec['mpi'].mpifc))
         else:
-            if '+debug' in self.spec:
-                extra_args.append('--enable-debug=all')
-            else:
-                extra_args.append('--enable-production')
+            args.append('-DHDF5_ENABLE_PARALLEL=OFF')
 
-            # '--enable-fortran2003' no longer exists as of version 1.10.0
-            if '+fortran' in self.spec:
-                extra_args.append('--enable-fortran2003')
-            else:
-                extra_args.append('--disable-fortran2003')
+        if '+static' in spec:
+            args.append('-DONLY_SHARED_LIBS=OFF')
+        else:
+            args.append('-DONLY_SHARED_LIBS=ON')
+
+        if '+debug' in spec:
+            args.append('-DCMAKE_BUILD_TYPE=Debug')
```

Comment:
Note, the `CMakePackage` declares variant `build_type` and sets '-DCMAKE_BUILD_TYPE` to a relevant value. I would say that we don't need variant `debug` for this package anymore.

Reply:
Since HDF5 has debug options orthogonal to the build type, maybe:
```python
args.append(self.define_from_variant('HDF5_ENABLE_TRACE', 'debug'))
```
?

Reply:
Removed.  HDF5_ENABLE_TRACE was considered inadvisable.